GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1952,8 +1952,8 @@ private void Stage(IReadOnlyList<GitItemStatus> items)
                                     return false;
                                 }
 
-                                Task<GitSubmoduleStatus?> statusTask = item.GetSubmoduleStatusAsync();
-
+                                Task<GitSubmoduleStatus?>? statusTask = item.GetSubmoduleStatusAsync();
+                                Validates.NotNull(statusTask);
                                 if (!statusTask.IsCompleted)
```

Comment:
I'm struggling with this - `statusTask` can be `null`, and in case when it is will it throw on line:1956 or line:1957?

Reply:
I rewrote most of the checks using pattern matching declarations, I find this much easier to read